### PR TITLE
Support for sending custom stack traces

### PR DESCRIPTION
**Pull request by redJ4y**
## Feature change:
- Add `CreateErrorWithStackTrace(message string, st StackTrace)` for sending a custom `raygun4go.StackTrace`. Inspired by @ramizpolic's [enable stacktrace overrides PR](https://github.com/MindscapeHQ/raygun4go/pull/35).
  - With accompanying documentation for GoDoc on `StackTrace` and `AddEntry` (for building custom stack traces).
  - With accompanying documentation in the readme.
  - With accompanying test.
  
## Tidying:
- General readme improvements.
  - Pull out broken Travis and gocover.io badges.
- Fix tests, particularly `raygun4go_test`:
  - Make integration test ability more clear. It's still partly manual, but at least that's more clear now!

### CODE REVIEW

File: raygun4go.go [hunk 1]
```diff
@@ -234,6 +235,19 @@ func (c *Client) CreateError(message string) error {
 	return c.Submit(post)
 }
 
+// Manually send an error to Raygun with a custom message and a custom stack trace.
```
Comment (QuantumNightmare) [inline raygun4go.go]: Other comments throughout this class have stacktrace as one word, so we should make that consistent.

File: raygun4go.go [hunk 2]
```diff
@@ -234,6 +235,19 @@ func (c *Client) CreateError(message string) error {
...
+// This is particularly useful in scenarios where the default system stack trace is not relevant or sufficient.
+//
+// message: The error message to be reported.
+// st:      A custom stack trace to be associated with the error.
+// Users can manually construct a StackTrace using the AddEntry method.
```
Comment (QuantumNightmare) [inline raygun4go.go]: Wondering if we want to mention where that AddEntry method is - or have an example of how to use it. The way it currently reads sounds like AddEntry is on this Raygun client.

**Review by QuantumNightmare — CHANGES_REQUESTED** [review]
Added some minor suggestions that I think would be good to sort out. Otherwise looks good as far as I can tell.

**Comment by ramizpolic** [comment]
My inbox got overrun @redJ4y, thanks for taking the initiative and working on this! Two things I wanted to address:
1. Is it necessary to bump the go version to accommodate these changes?
2. If so, could we consider splitting this into two PRs (first one being the feature, and the second one the cleanup)? This ensures that we can still use the feature without the version bump.

Note: I am not impact by any of these, it's just a consideration 😸

File: raygun4go.go (hunk 2 above)
Comment (ramizpolic) [reply raygun4go.go]: +1 
it would be useful if the actual stack trace type was documented and referenced in `CreateErrorWithStackTrace` docs

**Comment by redJ4y** [comment]
> Is it necessary to bump the go version to accommodate these changes?

Oops! After reviewing, the dependency update does not actually necessitate a Go 1.19 bump. I'll adjust the `go.mod` back to 1.13 and keep the PR focused on the feature changes! Thank you! 😄

File: go.sum [hunk 3]
```diff
@@ -1,15 +1,612 @@
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (QuantumNightmare) [inline go.sum:1]: This file is suddenly a whole lot bigger - I don't know if this is expected

**Review by QuantumNightmare — APPROVED** [review]
Thanks, just left a comment

File: go.sum (hunk 3 above)
Comment (redJ4y) [reply go.sum:1]: Should be all good! Just more checksums for dependencies to support newer versions of Go I believe. I have run `go mod tidy`, so I think this is as slim as it's going to get with Go 1.13 as the minimum version.

**Pull request merged**